addons/point_of_sale/static/src/app/screens/product_screen/product_screen.js

```diff
@@ -83,34 +83,43 @@ export class ProductScreen extends Component {
         });
         this.scrollDirection = useScrollDirection("products");
     }
-    getCategories() {
+    getAncestorsAndCurrent() {
         const selectedCategory = this.pos.selectedCategory;
-        const categoriesToDisplay = selectedCategory
-            ? [...selectedCategory.allParents, selectedCategory, ...selectedCategory.child_id]
+        return selectedCategory
+            ? [undefined, ...selectedCategory.allParents, selectedCategory]
+            : [selectedCategory];
+    }
+    getChildCategories(selectedCategory) {
+        return selectedCategory
+            ? [...selectedCategory.child_id]
             : this.pos.models["pos.category"].filter((category) => !category.parent_id);
-        return [
-            {
-                id: 0,
-                name: "",
-                icon: "fa-home fa-2x",
-            },
-            ...categoriesToDisplay.map((category) => {
-                return {
-                    ...pick(category, "id", "name", "color"),
-                    showSeparator:
-                        selectedCategory &&
-                        [
-                            ...selectedCategory.allParents.map((p) => p.id),
-                            this.pos.selectedCategory.id,
-                        ].includes(category.id),
-                    imgSrc:
-                        this.pos.config.show_category_images && category.has_image
-                            ? `/web/image?model=pos.category&field=image_128&id=${category.id}`
-                            : undefined,
-                };
-            }),
-        ];
     }
+    getChildCategoriesInfo(selectedCategory) {
+        return this.getCategoriesInfo(this.getChildCategories(selectedCategory), selectedCategory);
+    }
+    getCategoriesAndSub() {
+        return this.getAncestorsAndCurrent().flatMap((category) =>
+            this.getChildCategoriesInfo(category)
+        );
+    }
+
+    getCategoriesInfo(categories, selectedCategory) {
+        return categories.map((category) => {
```

Comment:
with map you can do 
```js
list.map(() => ({
  key1: 1,
}))
```
wrapping the `{}` in `()` allows you to not have to do a `return`